server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -680,59 +680,101 @@ private List<String> compilePodSourceFiles(PodSpec pod, Map<String, Object> mani
         List<String> objs = new ArrayList<>();
         List<String> commands = new ArrayList<>();
 
-        // per supported language build lists of flags and defines
+        // build lists of flags and defines per supported language
         List<String> defines = new ArrayList<>(pod.defines);
+        List<String> flagsC = new ArrayList<>(pod.flags);
         List<String> flagsCpp = new ArrayList<>(pod.flags);
         List<String> flagsObjC = new ArrayList<>(pod.flags);
-        List<String> flagsC = new ArrayList<>(pod.flags);
+        List<String> flagsObjCpp = new ArrayList<>(pod.flags);
         if (platform.contains("ios")) {
+            defines.addAll(pod.ios_defines);
+            flagsC.addAll(pod.ios_flags);
             flagsCpp.addAll(pod.ios_flags);
             flagsObjC.addAll(pod.ios_flags);
-            flagsC.addAll(pod.ios_flags);
-            defines.addAll(pod.ios_defines);
+            flagsObjCpp.addAll(pod.ios_flags);
         }
         else if (platform.contains("osx")) {
+            defines.addAll(pod.osx_defines);
+            flagsC.addAll(pod.osx_flags);
             flagsCpp.addAll(pod.osx_flags);
             flagsObjC.addAll(pod.osx_flags);
-            flagsC.addAll(pod.osx_flags);
-            defines.addAll(pod.osx_defines);
+            flagsObjCpp.addAll(pod.osx_flags);
         }
-        removeFlag("-std=c++", flagsObjC);
-        removeFlag("-std=c++", flagsC);
+
+        // TEMP FIX! remove hardcoded flags from context (should be removed from build.yaml perhaps?)
```

Comment:
Or maybe this isn't a temp fix? We probably only want to let pods toggle use of exceptions?